### ISSUE TITLE
Email address etc printing

### ISSUE THREAD

**Issue by cormacstrain** [cleidigh/printing-tools-ng#231]
Hi

ive selected the options to not print the header etc, but Im still getting the email info printing.  Is there a template or anything I can edit to modify how the final printed receipt looks?  I just want the content and subject of the email printing, not the email addresses etc

thanks

Cormac

**Comment by cleidigh** [comment]
@cormacstrain 
To only print the subject and body you only need to do the following : 
- On the headers option tab: 
- Set hide headers unchecked (meaning do not hide headers 
- in the headers order section, hide each header except Subject
- This will only show the subject and body

@cleidigh

**Comment by cormacstrain** [comment]
> @cormacstrain To only print the subject and body you only need to do the following :
> 
> * On the headers option tab:
> * Set hide headers unchecked (meaning do not hide headers
> * in the headers order section, hide each header except Subject
> * This will only show the subject and body
> 
> @cleidigh

hi - this works perfectly for emails that have already arrived and are in the inbox.  When I go to print, they work perfectly.

My issue is with emails that arrive and are then automatically printed out.  These emails ignore all the settings and include Subject, To, From etc.  Also, I have the font size set to 20 in the print setup, which as expected works fine with those emails that have already arrived in.  The font size doesnt work on those emails that are printing as they arrive.

So basically the settings are perfect if I choose to print an email after it has arrived.  If I choose to use the Filtaqulla plugin to have the email automatically print, it ignores the settings in PTNG

thanks

Cormac

**Comment by cleidigh** [comment]
@cormacstrain 
I forgot to ask about os, tb and ptng versions.
This doesn't seem right so I will try to reproduce once I know what you are using.
@cormacstrain

**Comment by cormacstrain** [comment]
@cleidigh

Running Ubuntu 22 using Thunderbird 102.13.0 and PTNG 2.2.3. the only other extension im using is FiltaQilla 3.7.2

thanks

**Comment by cormacstrain** [comment]
![3](https://github.com/cleidigh/printing-tools-ng/assets/60103489/152fd49e-c36b-4915-9495-dbdcffff2492)
It prints correctly (as above) when I select the email then send it to print.


![1](https://github.com/cleidigh/printing-tools-ng/assets/60103489/d739b195-15d8-45a5-858e-d8bc1433b248)
When the email comes in and automatically prints though, it seems to just ignore everything thats set in the prefs for PTNG and is like above - to/from/subject etc visible and the text much smaller

**Comment by cleidigh** [comment]
@cormacstrain 
Thanks for the images, they made me think that PTNG is not being invoked at all.
I have to ask are you sure FiltaQuila is set to use ptng for printing?

Also here is a test you can do to see debug output : 
- go to the console (Control-Shift-J)
- clear with trash can
- In ptng set the debug options to a single word, no spaces, lowecase
trace1

Now do an auto print.
Go back to the console, copy all and paste here. That should show us what is happening.
@cleidigh

**Comment by cormacstrain** [comment]
I set the debug and printed twice - the first from the email app directly and the second was the auto print.  I notice it says:'This page is in quirks mode. Page layout may be impacted' - would that have anything to do with it?


21:02:25.342 This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”. fetch>UID>.INBOX>542
21:02:29.188 PTNG: cmd_printng start: options : undefined printingtoolsng-pengine.js:587:12
21:02:29.204 PTNG: current window:
Object { id: 3, focused: true, top: 60, left: 26, width: 1257, height: 771, incognito: false, type: "normal", state: "normal", alwaysOnTop: false, … }
printingtoolsng-pengine.js:605:12
21:02:29.205 PTNG: current tab:
Object { id: 1, index: 0, windowId: 3, highlighted: true, active: true, status: "complete", width: 996, height: 344, url: "imap://food%40doubleu%2Eie@doubleu-ie-mail.dynu.com:993/fetch%3EUID%3E.INBOX%3E542", title: "Inbox", … }
printingtoolsng-pengine.js:606:12
21:02:29.206 PTNG: tab type : mail printingtoolsng-pengine.js:607:12
21:02:29.206 PTNG: tab url: imap://food%40doubleu%2Eie@doubleu-ie-mail.dynu.com:993/fetch%3EUID%3E.INBOX%3E542 printingtoolsng-pengine.js:608:12
21:02:29.207 PTNG: mailType: true printingtoolsng-pengine.js:609:12
21:02:29.207 PTNG: selectedMessageUris
Array [ "imap-message://food%40doubleu.ie@doubleu-ie-mail.dynu.com/INBOX#542" ]
printingtoolsng-pengine.js:63:12
21:02:29.214 PTNG: Use existing print hidden pane - multiple messages (1) printingtoolsng-pengine.js:336:12
21:02:29.215 PTNG: msgUris:
Array [ "imap-message://food%40doubleu.ie@doubleu-ie-mail.dynu.com/INBOX#542" ]
printingtoolsng-pengine.js:337:12
21:02:29.222 This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”. test.html
21:02:29.268 This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”. fetch>UID>.INBOX>542
21:02:30.282 PTNG: Done printingtoolsng-pengine.js:632:12
21:02:30.287 TypeError: browser is null BrowserElementParent.jsm:24:21
21:04:04.844 FiltaQuilla 21:4:4.844 [4435924 ms]
print
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), getProperty: getProperty(), setProperty: setProperty(), setStringProperty: setStringProperty(), getStringProperty: getStringProperty(), getUint32Property: getUint32Property(), setUint32Property: setUint32Property(), isRead: Getter, isFlagged: Getter, isKilled: Getter, … }
false filtaquilla-util.js:282:13
21:04:04.848 FiltaQuilla 21:4:4.848 [4 ms]
print.applyAction queue length: 1
Array [ XPCWrappedNative_NoHelper ]
filtaquilla-util.js:282:13
21:04:04.849 FiltaQuilla 21:4:4.849 [1 ms]
printNextMessage queue length: 1
Array [ XPCWrappedNative_NoHelper ]
filtaquilla-util.js:282:13
21:04:04.891 FiltaQuilla 21:4:4.891 [42 ms]
_printNextMessage(). Remaining queue length=0
XPCWrappedNative_NoHelper { QueryInterface: QueryInterface(), getProperty: getProperty(), setProperty: setProperty(), setStringProperty: setStringProperty(), getStringProperty: getStringProperty(), getUint32Property: getUint32Property(), setUint32Property: setUint32Property(), isRead: Getter, isFlagged: Getter, isKilled: Getter, … }
filtaquilla-util.js:282:13
21:04:04.894 Queue filter request to print message: [Pimento] New Order (2546)
21:04:05.438 gloda.datastore: got error in _asyncTrackerListener.handleError(): 19: constraint failed GlodaDatastore.jsm:1922:27
21:04:05.638 This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”. fetch>UID>.INBOX>572
21:04:05.676 TypeError: browser is null BrowserElementParent.jsm:24:21

**Comment by cormacstrain** [comment]
FiltaQuila is set to the 'print' option - it doesnt specifically say anything about ptng

**Comment by cleidigh** [comment]
@cormacstrain 
ok I don't think ptng is getting the print. In the FiltaQuila options you have to set the use PTNG for printing option 
@cleidigh

**Comment by cormacstrain** [comment]
silly me!  thanks a mil - that worked!

**Comment by cormacstrain** [comment]
only the best will do!  thanks again

**Comment by cormacstrain** [comment]
Im based in ireland

**Comment by cleidigh** [comment]
Been there multiple times and made good use of your Guinness...
Thanks and cheers!
Christopher

**Comment by cormacstrain** [comment]
let me know next time so I can buy you a few
